core/execution/market.go

```diff
@@ -774,6 +774,61 @@ func (m *Market) updateMarketValueProxy() {
 	m.stateChanged = true
 }
 
+func (m *Market) removeOrders(ctx context.Context) {
+	// remove all order from the book
+	// and send events with the stopped status
+	orders := append(m.matching.Settled(), m.peggedOrders.Settled()...)
+	orderEvents := make([]events.Event, 0, len(orders))
+	for _, v := range orders {
+		orderEvents = append(orderEvents, events.NewOrderEvent(ctx, v))
+	}
+
+	m.broker.SendBatch(orderEvents)
+}
+
+func (m *Market) clearAssetCollateral(ctx context.Context, mktState types.MarketState) error {
+	asset, _ := m.mkt.GetAsset()
+	parties := make([]string, 0, len(m.parties))
+	for k := range m.parties {
+		parties = append(parties, k)
+	}
+
+	sort.Strings(parties)
+	clearMarketTransfers, err := m.collateral.ClearMarket(ctx, m.GetID(), asset, parties)
+	if err != nil {
+		m.log.Error("Clear market error",
+			logging.MarketID(m.GetID()),
+			logging.Error(err))
+		return err
+	}
+
+	// unregister state-variables
+	m.stateVarEngine.UnregisterStateVariable(asset, m.mkt.ID)
+
+	m.broker.Send(events.NewTransferResponse(ctx, clearMarketTransfers))
+	m.mkt.State = mktState
+	m.mkt.TradingMode = types.MarketTradingModeNoTrading
+	m.broker.Send(events.NewMarketUpdatedEvent(ctx, *m.mkt))
+
+	return nil
+}
+
+func (m *Market) closeCancelledMarket(ctx context.Context, t time.Time) error {
+	m.tradableInstrument.Instrument.Unsubscribe(ctx)
+
+	err := m.clearAssetCollateral(ctx, types.MarketStateCancelled)
+	if err != nil {
+		return err
+	}
+
+	m.removeOrders(ctx)
```

Comment:
this doesn't make sense here, there won't be any settled orders, or any orders at all in the book

Reply:
You mean this should be removed ?

Reply:
based on what it's currently doing, I think so, yes.

Reply:
because I think after you're calling CancelAllOrders for all parties there would be nothing to do there.

Reply:
Yes, right.
I removed this https://github.com/vegaprotocol/vega/pull/5668/commits/e15634c60af3eeff4dd732ebb9f6404c0422e98e#diff-1782f5982916957ff0e7e82b0c7130c6d4c8ab4eb2e06f5c7b5b4a5925e15bcfR824
Thanks!